sechub-commons-model/src/test/java/com/mercedesbenz/sechub/commons/model/ModuleGroupTest.java

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.commons.model;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.EnumSource.Mode;
+import org.junit.jupiter.params.provider.NullSource;
+
+class ModuleGroupTest {
+
+    @ParameterizedTest
+    @EnumSource(ModuleGroup.class)
+    void resolveModuleGroupOrNull_for_the_scan_types_from_group_group_is_resolved(ModuleGroup groupToTest) {
+
+        for (ScanType type : groupToTest.moduleScanTypes) {
+            ModuleGroup resolvedGroup = ModuleGroup.resolveModuleGroupOrNull(Collections.singleton(type));
+            if (!(groupToTest.equals(resolvedGroup))) {
+                fail("Group:" + groupToTest + " should be resolved by scan type:" + type + " but resolved group was:" + resolvedGroup);
+            }
+        }
+    }
+
+    @Test
+    void resolveModuleGroupOrNull_for_null_returns_null() {
+        assertNull(ModuleGroup.resolveModuleGroupOrNull(null));
+    }
+
+    @Test
+    void resolveModuleGroupOrNull_for_empty_set_returns_null() {
+        assertNull(ModuleGroup.resolveModuleGroupOrNull(new LinkedHashSet<>()));
+    }
+
+    @Test
+    void resolveModuleGroupOrNull_for_report_returns_null() {
+        assertNull(ModuleGroup.resolveModuleGroupOrNull(Collections.singleton(ScanType.REPORT)));
+    }
+
+    @Test
+    void resolveModuleGroupOrNull_for_unknown_returns_null() {
+        assertNull(ModuleGroup.resolveModuleGroupOrNull(Collections.singleton(ScanType.UNKNOWN)));
+    }
+
+    @Test
+    void resolveModuleGroupOrNull_for_web_scan_returns_dynamic_group() {
+        assertEquals(ModuleGroup.DYNAMIC, ModuleGroup.resolveModuleGroupOrNull(Collections.singleton(ScanType.WEB_SCAN)));
+    }
+
+    /**
+     * It is forbidden that another group contains same scan type! We need the
+     * uniqueness.
+     */
+    @Test
+    void scan_type_is_assigned_only_for_one_group() {
+        /* prepare */
+        Map<ScanType, ModuleGroup> map = new LinkedHashMap<>();
+
+        /* test */
+        for (ScanType scanType : ScanType.values()) {
+
+            for (ModuleGroup group : ModuleGroup.values()) {
+                if (group.isForModulesHavingScanType(scanType)) {
```

Comment:
The object is an emum of type `ModuleGroup` and has the method `isForModulesHavingScanType`

So read this it is for me: this "module group is for modules having  scan type(givenType)"

`isAssignedBy(scanType)`
`isIdentifiedBy(scanType)`
`isRepresentingModulesOfScanType(scanType)`
`isContainingModulesOfScanType(scanType)`
`isRepresentingScanType(scanType)`
`isDeclaredByScanType(scanType)`

Please make a suggestion. I am happy with existing one and find it very readable/understandable.

Reply:
The problem is, that modules and scanType are basically the same thing, therefore Modules and ScanType in the same sentence is redundant.

From my perspective it would make sense to rename ScanType to Module or ModuleType. A scan from my perspective is: one execution of a configuration file (job). In addition, we should maybe create a glossary/definition of terms in the long run.

Follow up tasks:

1. Rename ScanType and ModuleType or Module
2. Issue create a glossary/definition of terms

My suggestion:

~~~
moduleGroup.isModuleInGroup(scanType)
~~~

Reading: Is the module x (scanType) in the module group?
Answer: Yes or No.

or 

~~~
moduleGroup.isGivenModuleInGroup(scanType)
~~~

Reading: Is the given module x (scanType) in the module Group?
Answer: Yes or No.

Reply:
I created
-  #1957 to handle `ScanType vs. Module`
- #1958 to update/create glossary

Reply:
I will use `isGivenModuleInGroup`